pkg/ipamclaimswebhook/podmutator.go

```diff
@@ -233,54 +302,117 @@ func ensureIPAMClaimRefAtNetworkSelectionElements(ctx context.Context,
 	return hasChangedNetworkSelectionElements, nil
 }
 
-func findNewPrimaryNetworkIPAMClaimName(ctx context.Context,
-	cli client.Client, pod *corev1.Pod, vmName string) (string, error) {
+func findPrimaryUDNLogicalNetworkAttachment(
+	ctx context.Context,
+	vmi *virtv1.VirtualMachineInstance,
+	pluginConfig *config.RelevantConfig,
+) (*virtv1.Interface, error) {
 	log := logf.FromContext(ctx)
-	if pod.Annotations[config.OVNPrimaryNetworkIPAMClaimAnnotation] != "" {
-		return "", nil
+
+	// TODO: once OVN-K lands this feature, we can remove uncomment this code
+	//if !pluginConfig.AllowPersistentIPs {
+	//	return nil, nil
+	//}
+
+	podNetwork := vmiPodNetwork(vmi)
```

Comment:
Don't we also want to check there is the "l2bridge" binding ? like we can end up returning a masquerade pod interface and that's wrong.

Reply:
Good point.

Reply:
fair enough, we can make it more picky at a follow up

Reply:
Resolving then; let's make a decision later.